torch/fft/__init__.py

```diff
@@ -9,18 +9,233 @@
+signal will be purely real.
+
+Note that some input frequencies must be purely real to satisfy the hermitian
+property. In these cases the imaginary component will be ignored.
+
+Args:
+    input (Tensor): the input tensor representing a half-hermitian signal
+    n (int, optional): Output signal length. This determines the length of the
+        output signal. If given, the input will either be zero-padded or trimmed to this
+        length before taking the IFFT. Defaults to odd output: ``n=input.size(dim) * 2 - 1``.
+    dim (int, optional): The dimension along which to take the one dimensional IFFT.
+    norm (str, optional): Normalization mode:
+        ``"forward"`` - the forward transform (:func:`~torch.fft.rfft`) is normalized by ``1/n``,
+        ``"backward"`` - the backward transform (:func:`~torch.fft.irfft`) is normalized by ``1/n``,
+        ``"ortho"`` - both transform directions are normalized by ``1/sqrt(n)`` (making it orthonormal).
+        Default is ``"backward"``.
+
+Example::
+
+    >>> import torch.fft
+    >>> t = torch.arange(4)
+    >>> t
+    tensor([0, 1, 2, 3])
+    >>> T = torch.fft.rfft(t)
+    >>> T
+    tensor([ 6.+0.j, -2.+2.j, -2.+0.j])
+
+    Without specifying the output length to :func:`torch.fft.irfft`, the output
+    will not round-trip properly because the input is even-length:
+
+    >>> torch.fft.irfft(T)
+    tensor([-0.4000,  0.8392,  1.1298,  2.0702,  2.3608])
+
+    So, it is recommended to always pass the signal length :attr:`n`:
+
+    >>> torch.fft.irfft(T, t.numel())
+    tensor([0., 1., 2., 3.])
+""")
+
+hfft = _add_docstr(_fft.fft_hfft, r"""
+hfft(input, n=None, dim=-1, norm=None) -> Tensor
+
+Computes the one dimensional discrete Fourier transform of a hermitian
+symmetric :attr:`input` signal.
+
+Because the signal is hermitian in the time-domain, the result will be real in
+the frequency domain. Note that some input frequencies must be purely real to
+satisfy the hermitian property. In these cases the imaginary component will be
+ignored.
+
+Args:
+    input (Tensor): the input tensor representing a half-hermitian signal
+    n (int, optional): Output signal length. This determines the length of the
+        real output. If given, the input will either be zero-padded or trimmed to this
+        length before taking the FFT. Defaults to odd output: ``n=input.size(dim) * 2 - 1``.
+    dim (int, optional): The dimension along which to take the one dimensional FFT.
+    norm (str, optional): Normalization mode:
+        ``"forward"`` - the forward transform (:func:`~torch.fft.hfft`) is normalized by ``1/n``,
+        ``"backward"`` - the backward transform (:func:`~torch.fft.ihfft`) is normalized by ``1/n``,
+        ``"ortho"`` - both transform directions are normalized by ``1/sqrt(n)`` (making it orthonormal).
+        Default is ``"backward"``.
+
+Example::
+
+    Taking a frequency signal that's purely real and bringing into the time
```

Comment:
If text is in an example it might render with code formatting, which could make it tricky to read. You could put the text inline like the other documentation text and then do `Example::` for the code snippets (possibly with comments). 

I could also be mistaken. If you try rendering this maybe it will render correctly.

Reply:
I built the docs, and removing the second colon made the text render as text and the doctest style code is still rendered as code.